03.bowling/frame.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+class Frame
+  attr_reader :shots
+
+  def initialize(shots, is_last_frame: false)
+    @shots = shots
+    @is_last_frame = is_last_frame
+  end
+
+  def score(index, all_frames)
+    base_score = @shots.sum(&:pins)
+    base_score = adjust_score_for_last_frame(base_score)
+    bonus = calculate_bonus(index, all_frames)
+    base_score + bonus
+  end
+
+  def strike?
+    @shots.first.strike?
+  end
+
+  def spare?
+    !strike? && @shots[0..1].sum(&:pins) == 10
+  end
+
+  private
+
+  def adjust_score_for_last_frame(base_score)
+    if @is_last_frame && @shots[0].strike?
+      base_score -= @shots[1..].sum(&:pins)
+    elsif @is_last_frame && (@shots.size > 2)
+      base_score -= @shots[2..].sum(&:pins)
+    end
+    base_score
+  end
+
+  def calculate_bonus(index, all_frames)
+    if strike?
+      strike_bonus(index, all_frames)
```

Comment:
```suggestion
      calculate_strike_bonus(index, all_frames)
```

メソッドはなるべく動詞で始めた方が親切かなーと思います

https://qiita.com/jnchito/items/459d58ba652bf4763820#%E5%87%A6%E7%90%86%E3%82%92%E5%AE%9F%E8%A1%8C%E3%81%99%E3%82%8B%E3%83%A1%E3%82%BD%E3%83%83%E3%83%89%E3%81%AF-%E5%8B%95%E8%A9%9E%E3%81%AE%E3%81%BF-or-%E5%8B%95%E8%A9%9E--%E5%90%8D%E8%A9%9E%E3%81%AB%E3%81%99%E3%82%8B